File: kotlin/programmers/level1/체육복.kt
class Solution {
    fun solution(n: Int, lost: IntArray, reserve: IntArray): Int {
        var answer = 0
        
        var arr = IntArray(n+2, {i -> 1})
        
        for(l in lost)
            arr[l] -= 1
        
        for(r in reserve)
            arr[r] += 1
        
        for(i in 1..n){
            if(arr[i] == 0 && arr[i-1] == 2){
                arr[i] += 1
                arr[i-1] -= 1
            } else if(arr[i]==0 && arr[i+1] == 2){
                arr[i] +=1
                arr[i+1] -= 1
            }
        }
        
        for(i in 1..n){
            if(arr[i] >= 1) answer += 1
        }
        
        return answer
    }
}

File: c++/programmers/level1/체육복.cpp
#include <vector>

using namespace std;

int solution(int n, vector<int> lost, vector<int> reserve) {
    int answer = 0;
    vector<int> v(n+2, 1);
    
    for(int i=0; i<reserve.size(); i++)
        v[reserve[i]]++;
    
    for(int i=0; i<lost.size(); i++)
        v[lost[i]]--;
    
    for(int i=1; i<=n; i++){
        if(v[i-1] == 0 && v[i] == 2) v[i-1] = v[i] = 1;
        else if(v[i] == 2 && v[i+1] == 0) v[i] = v[i+1] = 1;
    }
    for(int i=1; i<=n; i++) if(v[i] >= 1) answer++;
    
    return answer;
}